fdbserver/MoveKeys.actor.cpp

```diff
@@ -236,6 +236,45 @@ ACTOR Future<vector<UID>> addReadWriteDestinations(KeyRangeRef shard,
 	return result;
 }
 
+// Returns storage servers selected from 'candidates', who is serving a read-write copy of 'range'.
+ACTOR Future<std::vector<UID>> pickReadWriteServers(Transaction* tr, std::vector<UID> candidates, KeyRangeRef range) {
+	vector<Future<Optional<Value>>> serverListEntries;
+
+	for (const UID id : candidates) {
+		serverListEntries.push_back(tr->get(serverListKeyFor(id)));
+	}
+
+	vector<Optional<Value>> serverListValues = wait(getAll(serverListEntries));
+
+	std::vector<StorageServerInterface> ssis;
+	for (auto& v : serverListValues) {
+		ssis.push_back(decodeServerListValue(v.get()));
+	}
+
+	state std::vector<Future<Optional<UID>>> checks;
+	checks.reserve(ssis.size());
+	for (auto& ssi : ssis) {
+		checks.push_back(checkReadWrite(
+		    ssi.getShardState.getReplyUnlessFailedFor(GetShardStateRequest(range, GetShardStateRequest::NO_WAIT),
+		                                              SERVER_KNOBS->SERVER_READY_QUORUM_INTERVAL,
+		                                              0,
+		                                              TaskPriority::MoveKeys),
+		    ssi.id(),
+		    0));
+	}
+
+	wait(waitForAll(checks));
+
+	vector<UID> result;
+	for (const auto& it : checks) {
+		if (it.get().present()) {
+			result.push_back(it.get().get());
+		}
+	}
+
+	return result;
+}
+
 ACTOR Future<vector<vector<UID>>> additionalSources(RangeResult shards,
```

Comment:
our convention to improve comments in code is to  add comment for each actor a PR touch

Reply:
Done. Thanks for the info.